lib/portlayer/exec/exec.go

```diff
@@ -113,11 +113,14 @@ func Init(ctx context.Context, sess *session.Session, source extraconfig.DataSou
 func eventCallback(ie events.Event) {
 	// grab the container from the cache
 	container := Containers.Container(ie.Reference())
+
 	if container != nil {
+		container.m.Lock()
```

Comment:
yes. eventCallback may be called asynchronously wile container may be running another logic that could change the state. So, to insure there is no concurrent change of the state I need this lock here. This function doesn't run for a long time, so it will not block forever, just for several hundred nanoseconds.